init.lua

```diff
@@ -61,14 +65,45 @@ function filter.register_on_violation(func)
 	table.insert(filter.registered_on_violations, func)
 end
 
-function filter.check_message(name, message)
+-- Filters return a warning string if triggered
+function filter.register_filter(name, func)
```

Comment:
maybe `assert(!filter.registered_filters[name])` here to make sure people aren't adding two filters with the same name?

Reply:
Added